conda/common/configuration.py

```diff
@@ -0,0 +1,528 @@
+        super(ArgParseRawParameter, self).__init__(key, raw_value, None, None)
+
+    @classmethod
+    def make_raw_parameters(cls, args_from_argparse):
+        return super(ArgParseRawParameter, cls).make_raw_parameters(vars(args_from_argparse))
+
+
+class YamlRawParameter(RawParameter):
+    # this class should encapsulate all direct use of ruamel.yaml in this module
+
+    def __init__(self, ruamel_yaml_object, key):
+        rawvalue = ruamel_yaml_object[key]
+        keycomment = self._get_yaml_key_comment(ruamel_yaml_object, key)
+        keyflag = ParameterFlag.from_string(keycomment)
+        if isinstance(rawvalue, CommentedSeq):
+            valuecomments = self._get_yaml_list_comments(rawvalue)
+            valueflags = tuple(ParameterFlag.from_string(s) for s in valuecomments)
+            value = tuple(rawvalue)
+        elif isinstance(rawvalue, CommentedMap):
+            valuecomments = self._get_yaml_map_comments(rawvalue)
+            valueflags = dict((k, ParameterFlag.from_string(v))
+                              for k, v in iteritems(valuecomments) if v is not None)
+            value = dict(rawvalue)
+        elif isinstance(rawvalue, primitive_types):
+            valueflags = None
+            value = rawvalue
+        else:
+            raise ThisShouldNeverHappenError()  # pragma: no cover
+        super(YamlRawParameter, self).__init__(key, value, keyflag, valueflags)
+
+    @staticmethod
+    def _get_yaml_key_comment(commented_dict, key):
+        try:
+            return commented_dict.ca.items[key][2].value.strip()
+        except (AttributeError, KeyError):
+            return None
+
+    @staticmethod
+    def _get_yaml_list_comments(value):
+        items = value.ca.items
+        raw_comment_lines = tuple(excepts((AttributeError, KeyError, TypeError),
+                                          lambda q: items.get(q)[0].value.strip() or None,
+                                          lambda _: None  # default value on exception
+                                          )(q)
+                                  for q in range(len(value)))
+        return raw_comment_lines
+
+    @staticmethod
+    def _get_yaml_map_comments(rawvalue):
+        return dict((key, excepts(AttributeError,
+                                  lambda k: first(k.ca.items.values())[2].value.strip() or None,
+                                  lambda _: None  # default value on exception
+                                  )(key))
+                    for key in rawvalue)
+
+    @classmethod
+    def make_raw_parameters_from_file(cls, filepath):
+        with open(filepath, 'r') as fh:
+            ruamel_yaml = yaml_load(fh)
+        return cls.make_raw_parameters(ruamel_yaml)
+
+
+def load_raw_configs(search_path):
+    # returns an ordered map of filepath and dict of raw parameter objects
+
+    def _file_yaml_loader(fullpath):
+        assert fullpath.endswith(".yml") or fullpath.endswith("condarc"), fullpath
```

Comment:
Was this restriction necessary or documented? The switchover to Conda 4.2.6 yesterday broke the Conda.jl Julia wrappers because it was using the name `condarc-julia` - https://github.com/JuliaPy/Conda.jl/pull/51